homeassistant/components/evohome/__init__.py

```diff
@@ -141,9 +141,9 @@ async def authenticate(self, username: str, password: str) -> None:
             client_v2._user_account = None  # noqa: SLF001
 
         await client_v2.login()
-        await self.save_auth_tokens()
+        self.client_v2 = client_v2  # only set attr if authentication succeeded
 
-        self.client_v2 = client_v2
+        await self.save_auth_tokens()
```

Comment:
This is a bug, identified by the test. 

Is it ok to include in the same PR?

Reply:
awaiting #122664 to be merged...

Reply:
If the bug was merged to dev three days ago in a feature PR, the PR and bug haven't been released yet.

Reply:
all sorted.